File: Selenium/Activities/Activity8.py
from selenium import webdriver
from selenium.webdriver.support.wait import WebDriverWait
from selenium.webdriver.common.by import By
from selenium.webdriver.support import expected_conditions as EC

driver = webdriver.Chrome()
wait = WebDriverWait(driver, 10)

driver.get("https://v1.training-support.net/selenium/dynamic-controls")
print(driver.title)

togglebox = driver.find_element(By.ID, "toggleCheckbox")
checkbox = driver.find_element(By.ID, "dynamicCheckbox")

togglebox.click()
wait.until(EC.invisibility_of_element(checkbox))
print(f"check box is not visible {checkbox.text}")

togglebox.click()
wait.until(EC.visibility_of(checkbox))
print(checkbox.text)

checkbox.click()

driver.quit()

File: Selenium/Activities/Activity8.java
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.interactions.Actions;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.time.Duration;

public class Activity8 {
    public static void main(String[] args) {
        WebDriver driver = new FirefoxDriver();
        Actions builder = new Actions(driver);
        WebDriverWait wait =new WebDriverWait(driver, Duration.ofSeconds(1000));

        driver.get("https://v1.training-support.net/selenium/dynamic-controls");
        System.out.println("Page Title is " + driver.getTitle());

        WebElement toggleCheckbox = driver.findElement(By.id("toggleCheckbox"));
        WebElement dynamicBox = driver.findElement(By.id("dynamicCheckbox"));

        toggleCheckbox.click();
        wait.until(ExpectedConditions.invisibilityOf(dynamicBox));
        System.out.println("Dynamic check box is not visible " + dynamicBox.getText());

        toggleCheckbox.click();
        wait.until(ExpectedConditions.visibilityOf(dynamicBox));
        System.out.println(dynamicBox.getText());

        dynamicBox.click();

        driver.quit();
    }
}
